include/kassert/internal/logger.hpp

```diff
@@ -0,0 +1,150 @@
+// This file is part of KaMPI.ng.
+//
+// Copyright 2021-2022 The KaMPI.ng Authors
+//
+// KaMPI.ng is free software : you can redistribute it and/or modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later
+// version. KaMPI.ng is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the
+// implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+// for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License along with KaMPI.ng.  If not, see
+// <https://www.gnu.org/licenses/>.
+
+/// @file
+/// @brief Logger utility class to build error messages for failed assertins.
+
+#pragma once
+
+#include <ostream>
+#include <sstream>
+#include <type_traits>
+#include <utility>
+#include <vector>
+
+namespace kassert {
```

Comment:
Maybe `namespace kassert::internal {`?

Reply:
The idea here is that the user should overload `Logger::operator<<` for his own data types. Hence he could come into contact with this class, which is why it's not placed in `kamping::internal`